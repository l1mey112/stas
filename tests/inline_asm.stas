include <std.stas>

fn main {
	12
	20

	asm "rdi" "rsi" -> "rax" "rcx" {
		"add rdi, 8"
		"add rsi, 10"
		"mov rax, rdi"
		"mov rcx, rsi"
	}

	putuln
	putuln

	-1
	asm "rdi" -> "rcx" {
		"popcnt rcx, rdi"
	}
	putuln

	UINT32_MAX
	asm "rdi" -> "rdi" {
		"popcnt rdi, rdi"
	}
	putuln

	255
	asm "rdi" -> "rax" {
		"popcnt rax, rdi"
	}
	putuln
}