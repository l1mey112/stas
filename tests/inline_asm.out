30
20
64
32
8
